src/twin/ssh_tunnel.rs

```diff
@@ -0,0 +1,449 @@
+
+pub fn refresh_ssh_tunnel_status(_in_json: serde_json::Value) -> Result<Option<serde_json::Value>> {
+    twin::get_or_init(None).exec(|twin| twin.feature::<SshTunnel>()?.refresh_ssh_tunnel_status())
+}
+
+/// Get the public ssh key of the device
+///
+/// # Returns
+/// This function returns a json document containing the public key in PEM
+/// format. The document is formatted as follows:
+/// ```json
+/// {
+///   "pub_key": "..."
+/// }
+/// ```
+pub fn get_ssh_pub_key(_in_json: serde_json::Value) -> Result<Option<serde_json::Value>> {
+    twin::get_or_init(None).exec(|twin| twin.feature::<SshTunnel>()?.get_ssh_pub_key())
+}
+
+/// Instruct the device to establish an ssh tunnel to the bastion host
+///
+/// # Arguments
+/// * `args` - json object describing the properties of the ssh tunnel. This
+/// object must have the following structure:
+/// ```json
+/// {
+///   "tunnel_id": "...", // uuid identifying the tunnel
+///   "certificate": "..." // PEM formatted ssh certificate which the device uses to create the tunnel
+/// }
+/// ```
+///
+/// # Note
+/// This method is synchronous. It returns when the tunnel is created successfully.
+pub fn open_ssh_tunnel(in_json: serde_json::Value) -> Result<Option<serde_json::Value>> {
+    twin::get_or_init(None).exec(|twin| {
+        twin.feature::<SshTunnel>()?
+            .open_ssh_tunnel(in_json.to_owned())
+    })
+}
+
+/// Instruct the device to tear down an ssh tunnel to the bastion host
+///
+/// # Arguments
+/// * `args` - json object describing the ssh tunnel. This object must have the following structure:
+/// ```json
+/// {
+///   "tunnel_id": "...", // uuid identifying the tunnel
+/// }
+/// ```
+pub fn close_ssh_tunnel(in_json: serde_json::Value) -> Result<Option<serde_json::Value>> {
+    twin::get_or_init(None).exec(|twin| {
+        twin.feature::<SshTunnel>()?
+            .close_ssh_tunnel(in_json.to_owned())
+    })
+}
+
+static TUNNEL_TIMEOUT_SECS: usize = 30;
+
+static SSH_KEY_DIR: &str = "/home/omnect/.ssh";
+static SSH_KEY_NAME: &str = "bastion";
+static SSH_KEY_TYPE: &str = "ed25519";
+
+lazy_static::lazy_static! {
+    static ref CONTROL_SOCKET_DIR: String = env::var("XDG_RUNTIME_DIR").unwrap_or_else(|_| {
+        warn!("XDG_RUNTIME_DIR not set, defaulting to /tmp/omnect-device-service/sockets");
+        "/tmp/omnect-device-service/sockets".to_string()
+    });
+
+    static ref BASTION_SSH_USER: String = std::env::var("SSH_TUNNEL_USER").unwrap_or_else(|_| "ssh_user".to_string());
+    static ref BASTION_SSH_PORT: u16 = std::env::var("SSH_TUNNEL_PORT").map_or_else(|_| Ok(22),
+                                                                            |port| port.parse::<u16>()).unwrap();
+    static ref BASTION_HOST: String = std::env::var("BASTION_HOST").expect("BASTION_HOST must be set.");
+    static ref BASTION_SOCKET_DIR: String = std::env::var("BASTION_SOCKET_DIR").unwrap_or_else(|_| "/srv/ssh_sockets".to_string());
+}
+
+fn get_pub_key_path() -> PathBuf {
```

Comment:
I recently removed all `get_` prefixes as result of your review. Imho should not re-introduce here.